client/lib/i18n-utils/utils.js

```diff
@@ -149,7 +149,7 @@ export function removeLocaleFromPath( path ) {
  */
 export function getSupportLocale() {
 	const localeSlug = getLocaleSlug();
-	if ( config( 'support_locales' ).indexOf( localeSlug ) > -1 ) {
+	if ( config( 'support_site_locales' ).indexOf( localeSlug ) > -1 ) {
```

Comment:
Just like `support_site_locales` is much clearer than `support_locales`, how about we also rename `getSupportLocale()` as `getSupportSiteLocale()`? I think the later is also clearer :)